easybuild/tools/filetools.py

```diff
@@ -763,6 +763,18 @@ def apply_patch(patch_file, dest, fn=None, copy=False, level=None):
     apatch = os.path.abspath(patch_file)
     adest = os.path.abspath(dest)
 
+    if apatch.endswith(".gz"):
+        workdir = tempfile.mkdtemp(prefix='eb-patch-')
+        _, patchfilename_gz = os.path.split(apatch)
+        patchfilename, _ = os.path.splitext(patchfilename_gz)
+        dest_tmp = os.path.join(workdir, patchfilename)
+        _log.debug("Ungzipping the patch to: %s", workdir)
+
+        # gunzipping the patch.
+        extract_file(apatch, workdir)
+        # remove the '.gz' extension
+        apatch = dest_tmp
```

Comment:
`extract_file` returns the path to the extract file, no need to guess...

this simplifies this quite a bit:

```python
apatch = extract_file(apatch, tempfile.mkdtemp(prefix='eb-patch-'))
```